fileservice/filemaster/files.py

```diff
@@ -673,3 +676,39 @@ def proxy(self, request, uuid=None):
         log.debug(f'Sending user to S3 proxy: {response["X-Accel-Redirect"]}')
 
         return response
+
+
+class DownloadLogList(generics.ListAPIView):
+    """
+    A read-only endpoint for retreiving download logs.
+    """
+
+    serializer_class = DownloadLogSerializer
+
+    def get_queryset(self):
+        queryset = DownloadLog.objects.all()
+
+        user_email = self.request.query_params.get('user_email', None)
+        if user_email is not None:
+            queryset = queryset.filter(requesting_user__email=user_email)
+
+        uuid = self.request.query_params.get('uuid', None)
+        if uuid is not None:
+            queryset = queryset.filter(archivefile__uuid=uuid)
+
+        filename = self.request.query_params.get('filename', None)
+        if filename is not None:
+            queryset = queryset.filter(archivefile__filename=filename)
+
+        download_date_gte = self.request.query_params.get('download_date_gte', None)
+        if download_date_gte is not None:
+            queryset = queryset.filter(download_requested_on__gte=download_date_gte)
+
+        download_date_lte = self.request.query_params.get('download_date_lte', None)
+        if download_date_lte is not None:
+            queryset = queryset.filter(download_requested_on__lte=download_date_lte)
+
+        # Filter out files the user doesn't have permissions for.
+        queryset = [log for log in queryset if self.request.user.has_perm('filemaster.view_archivefile', log.archivefile)]
```

Comment:
Do a permissions check first and then apply other filters.  This for loop could take a long time if the number of logs requested is big

Reply:
@dtraviglia great idea -- just updated the PR with that in place